server/service/integration_enterprise_test.go

```diff
@@ -9420,6 +9420,9 @@ func (s *integrationEnterpriseTestSuite) TestSoftwareInstallerUploadDownloadAndD
 
 		// check activity
 		s.lastActivityOfTypeMatches(fleet.ActivityTypeDeletedSoftware{}.ActivityName(), fmt.Sprintf(`{"software_title": "ruby", "software_package": "ruby.deb", "team_name": "%s", "team_id": %d, "self_service": true}`, createTeamResp.Team.Name, createTeamResp.Team.ID), 0)
+
+		// download the installer, not found anymore
+		s.Do("GET", fmt.Sprintf("/api/latest/fleet/software/%d/package?alt=media", titleID), nil, http.StatusNotFound, "team_id", fmt.Sprintf("%d", *payload.TeamID))
```

Comment:
Note that this does not test the specific case where the installer exists in the DB but not in the storage (in this case it does not exist anywhere). This would require a relatively big refactor to get access to the storage manager and delete the installer from there without deleting it from the DB (it's created deep in the `SetupSuite`), and it's probably the only case where it would be useful. Given how trivial the fix was, I opted not to go down that path, but let me know if you think it'd be worth it.